cou/steps/plan.py

```diff
@@ -327,6 +329,8 @@ async def generate_plan(analysis_result: Analysis, args: CLIargs) -> UpgradePlan
     :rtype: UpgradePlan
     """
     pre_plan_sanity_checks(args, analysis_result)
+    hypervisors = await filter_hypervisors_machines(args, analysis_result)
+    logger.info("Hypervisors selected: %s", hypervisors)
```

Comment:
This should work only with hypervisors sub-command, or not?

Reply:
Not necessarily. The function abstracts the details and if nothing is provided, it will return all empty hypervisors and if force is used all hypervisors. Machines, hostnames and azs are just filters to create a sub-set of the hypervisors.

Reply:
If user use `cou upgrade controle-plane` we do not need to do this, so I think we should add one if condition to skip this in that case.

Reply:
Good catch. I'm thinking it would be better if we create different functions, which will be called by `generate_plan`, for generating plans for control-plane, data-plane, and hypervisors. Then we can move this hypervisor filter to the one(s) needing it.

Reply:
No need, since this will be not used only if user call `cou upgrade controle-plane`. In other cases, we will use it.

Reply:
We'll need to introduce the `hypervisor` command in a future PR and it's when we are going to fix this to not compute when using `control-plane`.

Because it's a dev branch, we decided to move forward to unblock the next PRs

Reply:
Ack :ok_hand: